test/Spec.hs

```diff
@@ -1,2 +1,62 @@
+import Test.Hspec
+
+import Marcher
+
+samplePoint :: Position
+samplePoint = Vec3 (0,0,0) -- Origin
+
+simpleScene :: Scene
+simpleScene = sphere (Vec3 (0, 0, (-3))) 1
+
+complexScene :: Scene
+complexScene = mergeScenes
+                  (colorize red $ sphere (Vec3 (0, 0, (-3))) 1)
+                  (colorize blue $ sphere (Vec3 (1, 1, (-2))) 0.1)
+
 main :: IO ()
-main = putStrLn "Test suite not yet implemented"
+main = hspec $ do
+  describe "Haskell Ray Marcher" $ do
+    describe "Function colorize" $ do
+      it "makes a scene blue" $ do
+        let (_, (color, _, _)) = colorize blue simpleScene samplePoint
+        color `shouldBe` blue
+
+      it "makes a scene red" $ do
+        let (_, (color, _, _)) = colorize red simpleScene samplePoint
+        color `shouldBe` red
+
+    describe "Function mixColors" $ do
+      it "mixes two colors additively" $ do
+        mixColors black white `shouldBe` gray -- black and white makes grey
```

Comment:
Could you please add all the tests that the documentation lists? That'd be great